test/unit-test/h264/h264_utest.c

```diff
@@ -205,18 +207,91 @@ void test_H264_Packetizer_AddFrame( void )
     result = H264Packetizer_Init( &( ctx ),
                                   &( nalusArray[ 0 ] ),
                                   MAX_NALUS_IN_A_FRAME );
+
+    TEST_ASSERT_EQUAL( H264_RESULT_OK,
+                       result );
+
+    frame.pFrameData = pFrame;
+    frame.frameDataLength = frameLength;
+
+    result = H264Packetizer_AddFrame( &( ctx ),
+                                      &( frame ) );
+
+    TEST_ASSERT_EQUAL( H264_RESULT_OK,
+                       result );
+
+    pkt.pPacketData = &( pktBuffer[ 0 ] );
+    pkt.packetDataLength = MAX_H264_PACKET_LENGTH;
+
+    result = H264Packetizer_GetPacket( &( ctx ),
+                                       &( pkt ) );
+
+    while( result != H264_RESULT_NO_MORE_PACKETS )
```

Comment:
Can we calculate the number of packet first? Then use the calculated number to check the expected result. For example, if we expect to have 3 packets from this frame, we can expect to have 3 OK returned and 1 NO_MORE_PACKETS at the end.